examples/anonymize_dataset.py

```diff
@@ -0,0 +1,203 @@
+write_anat(
+    image=t1w_path, bids_path=bids_path, landmarks=mri_landmarks, verbose=False
+)
+
+# %%
+# Basic anonymization
+# -------------------
+# Now we're ready to anonymize the dataset!
+
+anonymize_dataset(bids_root_in=bids_root, bids_root_out=bids_root_anon)
+
+# %%
+# That's it! Let's have a look at directory structure of the anonymized
+# dataset.
+print_dir_tree(bids_root_anon)
+
+# %%
+# You can see that the subject ID was changed to a number (in this case, the
+# digit ``1```), and the recording dates have been shifted backward in time (as
+# indicated by the ``emptyroom`` session name). Anonymized subject IDs are
+# simply a permutation of the set :math:`\{1, 2, \ldots, N\}``, where :math:`N`
+# is the total number of participants (excluding the ``emptyroom``
+# pseudo-subject). Zero-padding will be added to the subject IDs as needed to
+# allow for convenient sorting.
+#
+# Limiting to specific data types
+# -------------------------------
+# By default, :func:`mne_bids.anonymize_dataset` will anonymize
+# electrophysiological data and anatomical MR scans (T1-weightes and FLASH).
+# You can limit which data types to convert using the ``datatypes`` keyword
+# argument. The parameter can be a string (e.g., ``'meg'``, ``'eeg'``,
+# ``'anat'``) or a list of such strings.
+
+shutil.rmtree(bids_root_anon)
+anonymize_dataset(
+    bids_root_in=bids_root,
+    bids_root_out=bids_root_anon,
+    datatypes='anat'  # Only anatomical data
+)
+print_dir_tree(bids_root_anon)
+
+# %%
+# Specifying time shift
+# ---------------------
+# Anonymization involves shifting the recording dates back in time. MNE-BIDS
+# will try to automatically choose a suitable time shift. You may also
+# explicitly specify by how many days you wish to shift the recording dates
+# back in time via the ``daysback`` parameter. To avoid the time shift, pass
+# ``daysback=0``.
+
+shutil.rmtree(bids_root_anon)
+anonymize_dataset(
+    bids_root_in=bids_root,
+    bids_root_out=bids_root_anon,
+    datatypes='meg',  # Only MEG data
+    daysback=10
+)
+print_dir_tree(bids_root_anon / 'sub-emptyroom')  # Easy to see effects here
+
+# %%
+# Specifying subject IDs
+# ----------------------
+# Anonymized subject IDs are automatically generated as unique numbers in
+# ascending order. You can control this behavior via the ``subject_mapping``
+# parameter. Set it to ``None`` to avoid changing the subject IDs, e.g., in
+# case they're already anonymized. You can pass a dictionary that maps original
+# subject IDs to the anonymize IDs. Lastly, you can also pass a function that
+# accepts a list of original IDs and returns such a dictionary.
+
+shutil.rmtree(bids_root_anon)
+
+subject_mapping = {
+    'ABC123': 'anonymous',
+    'emptyroom': 'emptyroom'
+}
+
+anonymize_dataset(
+    bids_root_in=bids_root,
+    bids_root_out=bids_root_anon,
+    datatypes='meg',
+    subject_mapping=subject_mapping
+)
+print_dir_tree(bids_root_anon)
+
+# %%
+# Reproducibility
+# ---------------
```

Comment:
while I get the point of reproducibility, are you somehow recommending users to run the function every time the dataset changes? E.g., when a new subject is added. If so, question: what happens to padding in that case? Imagine you had 9 subjects and another subject was added. Now the old subjects will start having padding and it won't be reproducible any more

Reply:
No, incremental anonymization is not supported. If the input dataset changes, anonymization should be run on the entire dataset again. We sort-of enforce this by requiring that the output directory do not exist at the time `anonymize_dataset()` is called.

Reply:
I have now tried to clarify things in 246907a

Reply:
looks good!